repos/spack_repo/builtin/packages/erf/package.py

```diff
@@ -0,0 +1,174 @@
+    variant("fcompare", default=False, description="Enable fcompare")
+    variant("fft", default=False, description="Enable FFT support")
+    variant("debug", default=False, description="Enable debug build")
+
+    depends_on("cmake@3.20:", type="build")
+    depends_on("git", type="build")
+    depends_on("amrex", type=("build", "link", "run"))
+    for v in ("mpi", "openmp", "cuda", "sycl", "particles"):
+        depends_on(f"amrex+{v}", when=f"+{v}")
+        depends_on(f"amrex~{v}", when=f"~{v}")
+    for sm in CudaPackage.cuda_arch_values:
+        depends_on(f"amrex+cuda cuda_arch={sm}", when=f"+cuda cuda_arch={sm}")
+    depends_on("amrex+rocm", when="+hip")
+    depends_on("amrex~rocm", when="~hip")
+    depends_on("c", type="build")
+    depends_on("cxx", type="build")
+    depends_on("fortran", type="build")
+    depends_on("mpi", when="+mpi")
+    depends_on("cuda@11.0:", when="+cuda")
+    depends_on("hip@4.0:", when="+hip")
+    depends_on("rocm-openmp-extras", when="+hip")
+    depends_on("intel-oneapi-dpl", when="+sycl")
+    depends_on("intel-oneapi-mkl", when="+sycl")
+    depends_on("amrex@23.05:", when="@25.04")
+    depends_on("amrex@23.08:", when="@25.08")
+    depends_on("pkgconf", type="build")
+
+    with when("+netcdf"):
+        depends_on("netcdf-c+mpi+parallel-netcdf", when="+mpi")
+        depends_on("netcdf-c~mpi+parallel-netcdf", when="~mpi")
+        depends_on("netcdf-fortran")
+        depends_on("hdf5+mpi", when="+mpi")
+        depends_on("hdf5~mpi", when="~mpi")
+
+    depends_on("fftw", when="+fft")
+
+    conflicts("+cuda", when="+hip", msg="Cannot enable both CUDA and HIP")
+    conflicts("+openmp", when="+cuda", msg="Cannot enable both OpenMP and CUDA")
+    conflicts("+openmp", when="+hip", msg="Cannot enable both OpenMP and HIP")
+    conflicts("+cuda", when="+sycl", msg="Cannot enable both CUDA and SYCL")
+    conflicts("+hip", when="+sycl", msg="Cannot enable both HIP and SYCL")
+    conflicts("+fft", when="~mpi", msg="FFT support requires MPI")
+    conflicts("+radiation", when="platform=darwin", msg="Radiation is not supported on macOS")
+
+    def cmake_args(self):
+        args = [
+            self.define_from_variant("ERF_ENABLE_MPI", "mpi"),
+            self.define_from_variant("ERF_ENABLE_OMP", "openmp"),
+            self.define_from_variant("ERF_ENABLE_NETCDF", "netcdf"),
+            self.define_from_variant("ERF_ENABLE_PARTICLES", "particles"),
+            self.define_from_variant("ERF_ENABLE_MULTIBLOCK", "multiblock"),
+            self.define_from_variant("ERF_ENABLE_RADIATION", "radiation"),
+            self.define_from_variant("ERF_BUILD_TESTS", "tests"),
+            self.define_from_variant("ERF_BUILD_FCOMPARE", "fcompare"),
+            self.define_from_variant("ERF_ENABLE_FFT", "fft"),
+            self.define("ERF_DIM", "3"),
+        ]
+        args.append(self.define("ERF_USE_EXTERNAL_AMREX", False))
+        args.append(self.define("ERF_CLONE_AMREX", True))
```

Comment:
Please make Spack responsible for downloading amrex, rather than relying on the cmake system. This will make the package play much more nicely with spack features like source mirrors.